pkg/grpcutils/combine.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package grpcutils
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+)
+
+// CombineStreamServerInterceptors takes one or more grpc.StreamServerInterceptor and stacks them together into one.
+// Call order is left-to-right. The left-most interceptor is called first, with the right-most interceptor having the longest callstack.
+func CombineStreamServerInterceptors(interceptors ...grpc.StreamServerInterceptor) grpc.StreamServerInterceptor {
```

Comment:
When you look at the previous implementation of the combining only supported two interceptors. The code was much easier to follow.

https://github.com/storj/storj/pull/956/files#diff-2c40b511d93b3aa7d4cbb2a52314f28fL46

Reply:
isn't that identical to lines 25-29 here? i thought i was using that code.

given the new structure, it made sense to make it work with n-ary interceptors, which i suppose i could do as a separate function, but that's the brief switch statement at the beginning

Reply:
i guess i should say, that link is for UnaryServerInterceptors, and this comment is on StreamServerInterceptors. I also shortened the UnaryServerInterceptor code by one call

Reply:
But yeah, more of noting that this version is harder to read than the previous. Anyways, we can keep as is.